gammapy/cube/psf_map.py

```diff
@@ -0,0 +1,189 @@
+
+    # retrieve rad from psf IRF directly (this might be given as an argument instead)
+    rad = ref_geom.axes[0].center * ref_geom.axes[0].unit
+
+    # Compute separations with pointing position
+    separations = pointing.separation(ref_geom.to_image().get_coord().skycoord)
+    valid = np.where(separations < max_offset)
+
+    # Compute PSF values
+    psf_values = np.transpose(psf.evaluate(offset=separations[valid], energy=energy, rad=rad), axes=(2, 0, 1))
+    psfmap = Map.from_geom(ref_geom, unit='sr-1')
+    psfmap.data[:, :, valid[0], valid[1]] += psf_values.to(psfmap.unit).value
+    return psfmap
+
+
+class PSFMap():
+    """Class containing the Map of PSFs and allowing to interact with it.
+
+    Parameters
+    ----------
+    psf_map : `~gammapy.maps.Map`
+        the input PSF Map. Should be a Map with 2 non spatial axes.
+        rad and true energy axes should be given in this specific order.
+
+    """
+    def __init__(self, psf_map):
+        # Check the presence of an energy axis
+        if psf_map.geom.axes[1].type is not 'energy':
+            raise(ValueError,"Incorrect energy axis position in input Map")
+
+        if not u.Unit(psf_map.geom.axes[0].unit).is_equivalent('deg'):
+            raise(ValueError,"Incorrect rad axis position in input Map")
+
+        self._psfmap = psf_map
+        self._geom = psf_map.geom
+
+        self.energies = self._geom.axes[1].center * self._geom.axes[1].unit
+        self.rad = self._geom.axes[0].center * self._geom.axes[0].unit
+
+
+    @property
+    def psfmap(self):
+        """the PSFMap itself (~gammapy.maps.Map)"""
+        return self._psfmap
+
+    @classmethod
+    def from_file(cls, filename, **kwargs):
+        """ Read a psf_map from file and create a PSFMap object"""
+
+        psf_map.read(filename, **kwargs)
+        return cls(psf_map)
+
+    def get_energy_dependent_table_psf(self, position):
+        """ Returns EnergyDependentTable PSF at a given position
+
+        Parameters
+     ----------
+        position : `~astropy.coordinates.SkyCoord`
+            the target position. Should be a single coordinates
+
+        Returns
+        -------
+        psf_table : `~gammapy.irf.EnergyDependentTablePSF`
+            the table PSF
+        """
+        if position.size != 1:
+            raise ValueError("EnergyDependentTablePSF can be extracted at one single position only.")
+
+        # axes ordering fixed. Could be changed.
+        pix_ener = np.arange(self._geom.axes[1].nbin)
+        pix_rad = np.arange(self._geom.axes[0].nbin)
+
+        # Convert position to pixels
+        pix_lon, pix_lat = self._geom.to_image().coord_to_pix(position)
+
+        # Build the pixels tuple
+        pix = np.meshgrid(pix_lon, pix_lat,pix_rad,pix_ener)
+
+        # Interpolate in the PSF map. Squeeze to remove dimensions of length 1
+        psf_values = np.squeeze(self._psfmap.interp_by_pix(pix)*u.Unit(self._psfmap.unit))
```

Comment:
Do you need to `np.squeeze`, or could you explicitly index `[0]` or get rid of the extra dimension somehow?
We had `np.squeeze` a lot in the IRF code, but then there were cases where sometimes an axis only had one bin and `np.squeeze` removed that axis even though it was not supposed to.
So `np.squeeze` is IMO a little bit less explicit and error-prone compared to directly doing an index or reshape or transpose.

If you touch this, I'd also suggest to do the transpose here, and not in the constructor line  `EnergyDependentTablePSF (psf_value=psf_values.T)` below. I.e. prepare inputs for the call on the lines before. Again, IMO a bit easier to read, and also to debug in case one needs to enter a debugger, one directly sees the things and shapes that will be passed to the next function.